parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -32,20 +32,22 @@ pub use weights::WeightInfo;
 #[scale_info(skip_type_params(M))]
 pub struct Message<M: Get<u32>> {
 	/// ID of source parachain
-	parachain_id: ParaId,
+	origin: ParaId,
 	/// Unique nonce to prevent replaying messages
 	#[codec(compact)]
 	nonce: u64,
-	// TODO: add handler: u16?
+	/// Handler to dispatch the message to
+	handler: u16,
 	/// Payload for target application.
 	payload: BoundedVec<u8, M>,
 }
 
 impl<M: Get<u32>> Into<Token> for Message<M> {
 	fn into(self) -> Token {
 		Token::Tuple(vec![
-			Token::Bytes(self.parachain_id.encode()),
+			Token::Uint(u32::from(self.origin).into()),
```

Comment:
Had to add an explicit conversion here because `Uint` has a `From<u32>` but no `From<polkadot_parachain::primitives::Id>`